tests/slurm_plugin/test_capacity_block_manager.py

```diff
@@ -0,0 +1,745 @@
+                    ),
+                ],
+                [StaticNode("queue-cb-st-compute-resource-cb-1", "ip-1", "hostname-1", "some_state", "queue-cb")],
+                [True, True],
+                ["queue-cb-st-compute-resource-cb-1"],
+            ),
+            (
+                # return first nodename because there is an internal error
+                # in the second capacity block when creating Slurm reservation
+                True,
+                ["node1"],
+                {"cr-123456": CapacityBlock("cr-123456", "queue-cb", "compute-resource-cb")},
+                [
+                    CapacityReservationInfo(
+                        {**FAKE_CAPACITY_BLOCK_INFO, "State": "pending", "CapacityReservationId": "cr-123456"}
+                    )
+                ],
+                [StaticNode("queue-cb-st-compute-resource-cb-1", "ip-1", "hostname-1", "some_state", "queue-cb")],
+                [True, False],  # reservation creation failed for the first CB
+                ["queue-cb-st-compute-resource-cb-1"],
+            ),
+            (
+                # preserve original list of nodenames because there are internal errors
+                # while creating all the slurm reservations
+                True,
+                ["node1"],
+                {"cr-123456": CapacityBlock("cr-123456", "queue-cb", "compute-resource-cb")},
+                [
+                    CapacityReservationInfo(
+                        {**FAKE_CAPACITY_BLOCK_INFO, "State": "pending", "CapacityReservationId": "cr-123456"}
+                    ),
+                ],
+                [StaticNode("queue-cb-st-compute-resource-cb-1", "ip-1", "hostname-1", "some_state", "queue-cb")],
+                [False, False],  # reservation creation failed for all the CBs
+                ["node1"],
+            ),
+            (
+                # preserve old nodenames because there is a managed exception when contacting EC2
+                True,
+                ["node1"],
+                {"cr-123456": CapacityBlock("cr-123456", "queue-cb", "compute-resource-cb")},
+                AWSClientError("describe_capacity_reservations", "Boto3Error"),
+                [StaticNode("queue-cb-st-compute-resource-cb-1", "ip-1", "hostname-1", "some_state", "queue-cb")],
+                [False],
+                ["node1"],
+            ),
+            (
+                # preserve old nodenames because there is an unexpected exception when contacting EC2
+                True,
+                ["node1"],
+                {"cr-123456": CapacityBlock("cr-123456", "queue-cb", "compute-resource-cb")},
+                Exception("generic-error"),
+                [StaticNode("queue-cb-st-compute-resource-cb-1", "ip-1", "hostname-1", "some_state", "queue-cb")],
+                [False],
+                ["node1"],
+            ),
+            (
+                # return nodenames accordingly to the state (only pending)
```

Comment:
comment should be "only not active are returned", right?

Reply:
yes, fixed.